File: features/splash/src/java/kotlin/com.studio.splash/SplashViewModel.java
package com.studio.splash;

import android.Manifest;
import android.content.Context;

import com.studio.common.ui.base.BaseViewModel;
import com.studio.common.ui.constant.LocalConfigResourceData;
import com.studio.common.ui.utils.SingleLiveEvent;
import com.studio.permission.PermissionHelper;

import java.util.Arrays;
import java.util.List;
import java.util.Locale;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;
import dagger.hilt.android.qualifiers.ApplicationContext;
import timber.log.Timber;

@HiltViewModel
public class SplashViewModel extends BaseViewModel {

    private final Context mContext;
    private SingleLiveEvent<String> titleApp;
    private SingleLiveEvent<PermissionHelper> mPermission;

    @Inject
    public SplashViewModel(@ApplicationContext Context context, LocalConfigResourceData config) {
        mContext = context;
        String versionApp = mContext.getString(R.string.version_app, config.getAppName(), config.getVersionName());
        getTitleApp().postValue(versionApp);
    }

    public SingleLiveEvent<String> getTitleApp() {
        if (titleApp == null) {
            titleApp = new SingleLiveEvent<>();
        }
        return titleApp;
    }

    public SingleLiveEvent<PermissionHelper> getPermission() {
        if (mPermission == null) {
            mPermission = new SingleLiveEvent<>();
        }
        return mPermission;
    }

    public void setPermission(PermissionHelper permission) {
        getPermission().postValue(permission);
    }

    public void onSinglePermission() {
        getPermission().getValue().requestPermissions(Manifest.permission.POST_NOTIFICATIONS, Locale.getDefault().getLanguage(), () -> {
            Timber.e("--- Grant Permission success");
        }, deniedList -> {
            Timber.e("--- Permission denied: " + deniedList);
        }, permanentlyDeniedList -> {
            Timber.e("--- Permission permanentlyDeniedList: " + permanentlyDeniedList);
        });
    }

    public void onDualPermission() {
        List<String> permissions = Arrays.asList(Manifest.permission.POST_NOTIFICATIONS, Manifest.permission.NEARBY_WIFI_DEVICES);
        getPermission().getValue().requestPermissions(permissions, Locale.getDefault().getLanguage(), () -> {
            Timber.e("--- Grant Permission success");
        }, deniedList -> {
            Timber.e("--- Permission denied: " + deniedList);
        }, permanentlyDeniedList -> {
            Timber.e("--- Permission permanentlyDeniedList: " + permanentlyDeniedList);
        });
    }

    public void onTriplePermission() {
        List<String> permissions = Arrays.asList(Manifest.permission.POST_NOTIFICATIONS, Manifest.permission.NEARBY_WIFI_DEVICES, Manifest.permission.CAMERA, Manifest.permission.READ_SMS, Manifest.permission.RECEIVE_SMS);
        getPermission().getValue().requestPermissions(permissions, Locale.getDefault().getLanguage(), () -> {
            Timber.e("--- Grant Permission success");
        });
    }
}

File: features/splash/src/main/java/com/studio/splash/SplashViewModel.kt
package com.studio.splash

import android.Manifest
import android.content.Context
import androidx.activity.ComponentActivity
import androidx.lifecycle.viewModelScope
import com.studio.common.ui.base.BaseViewModel
import com.studio.common.ui.constant.LocalConfigResourceData
import com.studio.common.ui.utils.SingleLiveEvent
import com.studio.permission.PermissionHelper
import dagger.hilt.android.lifecycle.HiltViewModel
import dagger.hilt.android.qualifiers.ApplicationContext
import kotlinx.coroutines.flow.launchIn
import kotlinx.coroutines.launch
import timber.log.Timber
import java.util.Locale
import javax.inject.Inject

@HiltViewModel
class SplashViewModel
    @Inject
    constructor(
        @ApplicationContext val context: Context,
        config: LocalConfigResourceData
    ) : BaseViewModel() {
        val titleApp by lazy { SingleLiveEvent<String>() }

        init {
            val versionApp =
                context.resources.getString(R.string.version_app, config.appName, config.versionName)
            titleApp.postValue(versionApp)
        }

        fun onSinglePermission(activity: ComponentActivity) {
            viewModelScope.launch {
                PermissionHelper(activity)
                    .requestPermissions(
                        Manifest.permission.POST_NOTIFICATIONS,
                        language = Locale.getDefault().language,
                        onGranted = {
                            Timber.e("--- Grant Permission success")
                        },
                        onDenied = {
                            Timber.e("--- Grant Permission onDenied")
                        },
                        onPermanentlyDenied = {
                            Timber.e("--- Grant Permission onPermanentlyDenied")
                        }
                    ).launchIn(this)
            }
        }

        fun onDualPermission(activity: ComponentActivity) {
            viewModelScope.launch {
                PermissionHelper(activity)
                    .requestPermissions(
                        Manifest.permission.POST_NOTIFICATIONS,
                        Manifest.permission.NEARBY_WIFI_DEVICES,
                        language = Locale.getDefault().language,
                        onGranted = {
                            Timber.e("--- Grant Permission success")
                        },
                        onDenied = {
                            Timber.e("--- Grant Permission onDenied")
                        },
                        onPermanentlyDenied = {
                            Timber.e("--- Grant Permission onPermanentlyDenied")
                        }
                    ).launchIn(this)
            }
        }

        fun onTriplePermission(activity: ComponentActivity) {
            viewModelScope.launch {
                PermissionHelper(activity)
                    .requestPermissions(
                        Manifest.permission.POST_NOTIFICATIONS,
                        Manifest.permission.CAMERA,
                        Manifest.permission.READ_SMS,
                        language = Locale.getDefault().language,
                        onGranted = {
                            Timber.e("--- Grant Permission success")
                        }
                    ).launchIn(this)
            }
        }
    }
